src/middleware.rs

```diff
@@ -164,3 +164,67 @@ where
         (self)(request, next)
     }
 }
+
+#[cfg(feature = "digest-auth")]
+pub mod digest {
+    use super::*;
+    use crate::{Request, Response};
+    use digest_auth::{AuthContext, WwwAuthenticateHeader};
+    use std::{borrow::Cow, str::FromStr};
+
+    /// Provides simple digest authentication powered by the `digest_auth` crate.
+    ///
+    /// Any 401 response handled by this middleware is retried once with the
+    /// credentials provided on construction. If authentication fails or the middleware
+    /// is unable to generate an answer to the server challenge (such as a different authentication
+    /// scheme or a malformed challenge) the response is silently passed on.
```

Comment:
I always appreciate rustdoc that comes with an example code :) The invocation is super simple, so I don't feel strongly, though.

Reply:
Done! :)